src/jobs_done10/generators/jenkins.py

```diff
@@ -758,12 +758,19 @@ def _GetJenkinsJobBranch(self, jenkins_api, jenkins_job):
         scms = root.findall('scm/scms/hudson.plugins.git.GitSCM')
 
         checked_urls = []
+
+        # Account for case-differences
+        repo_lower = self.repository.url.lower()
```

Comment:
Yes. We had a bunch of release candidates and alpha versions for the `4.2.0` release, and when it came time to push the main `rb-ROCKY-v4.2.0` branch job creation failed:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/server.py", line 28, in index
    return process_jobs_done(data)
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/server.py", line 139, in process_jobs_done
    password=jenkins_password,
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/generators/jenkins.py", line 810, in UploadJobsFromFile
    return publisher.PublishToUrl(url, username, password)
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/generators/jenkins.py", line 650, in PublishToUrl
    matching_jobs = self._GetMatchingJobs(jenkins_api)
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/generators/jenkins.py", line 720, in _GetMatchingJobs
    jenkins_job_branch = self._GetJenkinsJobBranch(jenkins_api, jenkins_job)
  File "/usr/local/lib/python3.6/site-packages/jobs_done10/generators/jenkins.py", line 779, in _GetJenkinsJobBranch
    raise RuntimeError('\n'.join(error_msg))
RuntimeError: Could not find SCM for repository "ssh://git@eden.fln.esss.com.br:7999/esss/rocky30.git" in job "rocky30-rb-ROCKY-v4.2.0-rc.2-linux64-27"
The local repository origin is set to "ssh://git@eden.fln.esss.com.br:7999/esss/rocky30.git".
And the possible matches are:
 - ssh://git@eden.fln.esss.com.br:7999/ESSS/rocky30
 - ssh://git@eden.fln.esss.com.br:7999/esss/eden.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/ben10.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/etk.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/kraken20.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/rocky20_examples.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/rocky20.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/rocky30scheduler.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/rocky.git
 - ssh://git@eden.fln.esss.com.br:7999/esss/rocky_docs.git

If needed a repository origin url can be set with "git remote set-url origin <URL>".
```

This happened while jobs_done was parsing all jobs with the `rb-ROCKY-v4.2.0`. This includes `rb-ROCKY-v4.2.0-rc.2`, which is somehow configured to have the remote git url `ssh://git@eden.fln.esss.com.br:7999/ESSS/rocky30`. I don't know how that happened. So that failed both due to the lack of the `.git` suffix and the uppercase `ESSS`. The changes that this commit makes do fix this particular issue we had.

Reply:
I did see those paths in the past because the job was originaly created from the command line where the root project's repo remote does not match the expected pattern (much like your case here).

Reply:
Yes, looks like it depends on how you do the initial cloning, which is unfortunate